servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/FromInputStreamPublisher.java

```diff
@@ -62,14 +63,29 @@
     private volatile int subscribed;
 
     private final InputStream stream;
+    private final int readChunkSize;
 
     /**
      * A new instance.
      *
      * @param stream the {@link InputStream} to expose as a {@link Publisher}
      */
     FromInputStreamPublisher(final InputStream stream) {
+        this(stream, DEFAULT_READ_CHUNK_SIZE);
+    }
+
+    /**
+     * A new instance.
+     *
+     * @param stream the {@link InputStream} to expose as a {@link Publisher}
+     * @param readChunkSize the maximum length of {@code byte[]} chunks which will be read from the {@link InputStream}
+     * and emitted by the {@link Publisher}.
+     */
+    FromInputStreamPublisher(final InputStream stream, final int readChunkSize) {
         this.stream = requireNonNull(stream);
+        if (readChunkSize <= 0) {
+            throw new IllegalArgumentException("readChunkSize: " + readChunkSize + " (expected: >0)");
+        }
```

Comment:
🤦‍♂️ need some practice with GH suggestions
```suggestion
        }
        this.readChunkSize = readChunkSize;
```